src/preact/compat.js

```diff
@@ -5,23 +5,24 @@ import * as mode from '#core/mode';
 const REACT_FORWARD_SYMBOL =
   (typeof Symbol !== 'undefined' && Symbol.for?.('react.forward_ref')) || 0xf47;
 
-// `__b` is the known mangled name exported by preact. This is called during
-// the diffing algorithm with the constructed VNode, before the vnode is
-// actually used to diff.
-const oldDiff = options['__b'];
-options['__b'] = newDiff;
+/**
+ * @type {*}
+ * `__b` is the known mangled name exported by preact. This is called during
+ * the diffing algorithm with the constructed VNode, before the vnode is
+ * actually used to diff.
+ *
+ */
+const diffKey = '__b';
+const oldDiff = /** @type {*} */ (options)[diffKey];
+/** @type {*} */ (options)[diffKey] = newDiff;
 
 /**
  * Checks if our VNode type has a `forwardRef_` sigil, in which case it was created by our forwardRef implementation.
  * If so, we move the VNode's ref to it's props, which will be passed to the Component as its props.
  *
  * See VNode type at https://github.com/preactjs/preact/blob/55ee4bc069f84cf5e3af8f2e2f338f4e74aca4d8/src/create-element.js#L58-L76
  *
- * @param {{
- *   type: (string|function()),
- *   ref: *,
- *   props: !Object,
- * }} vnode
+ * @param {*} vnode
```

Comment:
Should this really be any?

Reply:
See my earlier comment (https://github.com/ampproject/amphtml/pull/37322#discussion_r782323929). I don't really care about this file. Prev types were incorrect, and throwing some `any` into here is better than removing it from type checking.

Reply:
SGTM